R/derive_date_vars.R

```diff
@@ -427,20 +439,35 @@ compute_dtf <- function(dtc, dt) {
 #' compute_tmf(dtc = "2019-07-18T15:25", dtm = as.POSIXct("2019-07-18T15:25:00"))
 #' compute_tmf(dtc = "2019-07-18T15", dtm = as.POSIXct("2019-07-18T15:25:00"))
 #' compute_tmf(dtc = "2019-07-18", dtm = as.POSIXct("2019-07-18"))
-compute_tmf <- function(dtc, dtm) {
+compute_tmf <- function(dtc,
+                        dtm,
+                        ignore_seconds_flag = FALSE) {
+
   assert_that(is.character(dtc), is_date(dtm))
+  assert_logical_scalar(ignore_seconds_flag)
 
   is_na <- is.na(dtm)
   n_chr <- nchar(dtc)
   valid_dtc <- is_valid_dtc(dtc)
   warn_if_invalid_dtc(dtc, valid_dtc)
 
-  case_when(
-    (!is_na & n_chr >= 19 & valid_dtc) | is_na | !valid_dtc ~ NA_character_,
-    n_chr == 16 ~ "S",
-    n_chr == 13 ~ "M",
-    n_chr == 10 | (n_chr > 0 & n_chr < 10) ~ "H"
-  )
+
+  if (ignore_seconds_flag)  {
+    if ((any(n_chr >= 17)))
+    {abort("Seconds detected in data while ignore_seconds_flag is invoked")
+    }
+    else {
+      case_when(
+        (!is_na & n_chr >= 19 & valid_dtc) | is_na | !valid_dtc ~ NA_character_,
+        n_chr == 13 ~ "M",
+        n_chr == 10 | (n_chr > 0 & n_chr < 10) ~ "H"
+      )}
+  } else {
+    case_when(
+      (!is_na & n_chr >= 19 & valid_dtc) | is_na | !valid_dtc ~ NA_character_,
+      n_chr == 16 ~ "S",
+      n_chr == 13 ~ "M",
+      n_chr == 10 | (n_chr > 0 & n_chr < 10) ~ "H")}
```

Comment:
lintr: "Closing curly-braces should always be on their own line, unless it's followed by an else."